src/context/AppContextProvider.tsx

```
@@ -72,75 +72,86 @@ export const AppContextProvider = ({
+        // start the network switching process
+        if (
+          operation.walletMustMatchChainID &&
+          walletStore.getSnapshot().walletType === WalletTypes.METAMASK &&
+          walletStore.getSnapshot().walletChainId !== chainId
+        ) {
+          await walletStore.metamaskSwitchNetwork(chainName);
         }
-      }
 
-      // if the chain id in metamask doesn't match the chain id we need to be on
-      // start the network switching process
-      if (
-        operation.walletMustMatchChainID &&
-        walletStore.getSnapshot().walletType === WalletTypes.METAMASK &&
-        walletStore.getSnapshot().walletChainId !== chainId
-      ) {
-        await walletStore.metamaskSwitchNetwork(chainName);
-      }
+        if (!operation.validate(params, walletStore)) {
+          throw new Error('Invalid parameters for operation');
+        }
 
-      if (!operation.validate(params, walletStore)) {
-        throw new Error('Invalid parameters for operation');
-      }
+        if ('buildTransaction' in operation) {
+          return (operation as ChainMessage<unknown>).buildTransaction(
+            params,
+            walletStore,
+          );
+        } else if ('executeQuery' in operation) {
+          return (operation as ChainQuery<unknown>).executeQuery(
+            params,
+            walletStore,
+          );
+        }
 
-      if ('buildTransaction' in operation) {
-        return (operation as ChainMessage<unknown>).buildTransaction(
-          params,
-          walletStore,
-        );
-      } else if ('executeQuery' in operation) {
-        return (operation as ChainQuery<unknown>).executeQuery(
-          params,
-          walletStore,
+        throw new Error('Invalid operation type');
+      } catch (error) {
+        setErrorText(
+          'An error occured: '.concat(
+            error instanceof Error
+              ? error.message
+              : 'An unknown error occurred',
+          ),
```

Set the error (thrown from within the validate method) in state so we can render it

---

shouldn't the model get the error and show it to the user? i don't think this is ideal, we need to complete the tool call, now if an error occurs the model will not be aware.

---

we just need to figure out why what we had before broke, i would expect the thrown error to be handled in `App.tsx` like it currently is 
```
        const result = await executeOperation(
          name,
          toolCall.function.arguments,
        );
        content = JSON.stringify({
          status: 'ok',
          info: result,
        } as OperationResult);
      } catch (err) {
        content = JSON.stringify({
          status: 'failed',
          info: err,
        } as OperationResult);
```

---

if we catch the error inside executeOperation it will be swallowed and the model will not get a response for it's tool call